### DISCUSSION TITLE
Lighthouse deck using custom hardware issues

### DISCUSSION THREAD

**Discussion by abeinder** [bitcraze/discussions#683] [Q&A] [answered]
I am attempting to use some of the scripts in the lighthouse-bootloader and lighthouse-fpga repositories in order to boot the lighthouse deck up and read beam angles on a custom project (we are putting a lighthouse deck on an zybo controlled roomba for an embedded systems final project). I am having trouble getting the lighthouse deck to do much of anything. Using the uart_bootloader.py script I am able to flash the v6 firmware binary copied directly from the latest release onto the deck, and I am also able to read a chip id. I can also confirm that the V6 firmware is on there because the uart_bootloader.py script now says the firmware is unchanged. I have a voltage divider on my input to the rx line on the deck as well, and I have an FTDI. However, if I use the decodeV2 or decodeV2_cf scripts I am unable to read anything, I hang on the src.read line for the sync.

Also, when I boot up using the 0xBC followed by 0x00 boot operation, the blue light on the crazyflie comes on but nothing else. On the crazyflie itself, I notice a yellow light followed by a green light, though this might be from the crazyflie firmware itself as setting the lights appears to be a command I should be able to send. However, I also am unable to set the lights using the 0x01 0x## command. I use 0x01 0x01 to see if it does anything. I have noticed that sometimes if I send a really bad command the orange light will come on.

Hooking up to an oscilloscope, I do not see any data being transmitted from the deck for a "sync" packet, which if booted correctly I would expect to see 1-2 times a second. I have confirmed that the lighthouse deck is not fried by connecting it to a crazyflie and it boots up correctly and outputs angles because the crazyflie is listing positions in cfclient, and the light turns from yellow to green in the presence of sweep angles.

I have been perusing the forum and website and repositories for a while now and don't think I'm going to figure this out anytime soon. Any help would be greatly appreciated!

PS: These threads have been helpful, but I'm not sure they fully answered my questions:
a. https://forum.bitcraze.io/viewtopic.php?t=4223
b. https://forum.bitcraze.io/viewtopic.php?t=4961
c. https://forum.bitcraze.io/viewtopic.php?t=4400
Perhaps one or two other conversations as well

**Comment by ataffanel** [comment, marked as answer]
Hi,

First of all what I would expect: if the deck is already working on a Crazyflie, you should be able to run the [bootloader script](https://github.com/bitcraze/lighthouse-bootloader/blob/master/scripts/uart_bootloader.py) and it will leave the deck in working state: when the script is done, the deck is booted and it should start sending syncs on the UART. The proper way to use the deck from a raspberypi or equivalent would be to run this script at startup and then to receive data from the deck.

Are you using the same UART in both cases? While the bootloader works on the external uart (the one exposed on soldering pads), the firmware only works with the deck-uart, the same connected to a Crazyflie.

You are correct about the LEDs: they are controlled from the Crazyflie using the command 0x01. The protocol is documented in the [fpga repos](https://github.com/bitcraze/lighthouse-fpga#uart-protocol). There should be no need to send 0xBC all the time by the way. Like on the [bootloader script](https://github.com/bitcraze/lighthouse-bootloader/blob/master/scripts/uart_bootloader.py#L9), a UART break and 0xBC only needs to be sent once to synchronize the communication state machine. After that, as long as your communicaiton is correctly implemented, you can just send commands directly.

**Reply by abeinder** [reply]
Thank you, this worked immediately. I thought that I had probed the uart pins but apparently not.
